### PR TITLE
Added IsInteger, IsIntegerLaneType, RemoveVolatile, and RemoveCvRef

### PR DESCRIPTION
**Pull request by johnplatts**
Added the following type traits:
- `IsInteger<T>()` - returns true if `T` is one of the following types and returns false otherwise:
  `bool`, `char`, `signed char`, `unsigned char`, `short`, `unsigned short`, `int`, `unsigned int`, `long`, `unsigned long`, `long long`, `unsigned long long`, `int8_t`, `uint8_t`, `int16_t`, `uint16_t`, `int32_t`, `uint32_t`, `int64_t`, `uint64_t`, `char8_t`, `char16_t`, `char32_t`, `wchar_t`, `intptr_t`, `uintptr_t`, `size_t`, `ptrdiff_t`
- `IsIntegerLaneType<T>()` - returns true if `T` is one of the following types and returns false otherwise:
  `int8_t`, `uint8_t`, `int16_t`, `uint16_t`, `int32_t`, `uint32_t`, `int64_t`, `uint64_t`
- `RemoveVolatile<T>` - removes the `volatile` qualifier off of `T`
- `RemoveCvRef<T>` - equivalent to `RemoveConst<RemoveVolatile<RemoveRef<T>>>`

Resolves issue #1727.

### CODE REVIEW

File: hwy/base.h [hunk 1]
```diff
@@ -1174,14 +1299,14 @@ constexpr bool IsSigned<bfloat16_t>() {
 // Largest/smallest representable integer values.
 template <typename T>
 HWY_API constexpr T LimitsMax() {
-  static_assert(!IsFloat<T>(), "Only for integer types");
-  using TU = MakeUnsigned<T>;
-  return static_cast<T>(IsSigned<T>() ? (static_cast<TU>(~0ull) >> 1)
-                                      : static_cast<TU>(~0ull));
+  static_assert(IsInteger<T>(), "Only for integer types");
+  using TU = UnsignedFromSize<sizeof(T)>;
```
Comment (jan-wassenberg) [inline hwy/base.h:1303]: Should we change MakeUnsigned to this implementation?

**Review by jan-wassenberg — APPROVED** [review]
Very nice, thanks for adding this and the extra checks :)

File: hwy/base.h (hunk 1 above)
Comment (johnplatts) [reply hwy/base.h:1303]: The reason why `UnsignedFromSize<sizeof(T)>` is used instead of `MakeSigned<T>` is that it is possible for `!IsSame<size_t, uint32_t>() && !IsSame<size_t, uint64_t>()` to be true on some platforms and the current implementation of `hwy::MakeSigned<T>` will give compiler errors if `!IsIntegerLaneType<T>() && !IsFloat<T>() && !IsSpecialFloat<T>()` is true.

Comment (jan-wassenberg) [reply hwy/base.h:1303]: I understand why you are using UnsignedFromSize, this makes sense. Let me rephrase: what if we changed the implementation of Make(Un)Signed to what you are doing here, then we could still use Make(Un)Signed here, because it is a bit shorter/more clear?
